### PR TITLE
fix: fix lag/lead default value when column is nullable

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR
1. fix lag/lead default value when the column is nullable
2. fix count(*) can be used in window function
3. add more window function tests that derived from duckdb
4. fix lag function with partition by

Closes #issue

### CODE REVIEW

File: src/query/expression/src/values.rs [hunk 1]
```diff
@@ -2443,7 +2443,11 @@ impl ColumnBuilder {
                     field.append_column(other_field);
                 }
             }
-            (this, other) => unreachable!("unable append {other:?} into {this:?}"),
+            (this, other) => unreachable!(
```
Comment (sundy-li) [inline src/query/expression/src/values.rs:2446]: there are another two places.

File: src/query/service/src/pipelines/processors/transforms/window/transform_window.rs [hunk 2]
```diff
@@ -490,28 +490,29 @@ impl<T: Number> TransformWindow<T> {
...
-                        col.index(self.frame_start.row).unwrap().to_owned()
+                let lag_row = if self.current_row.block == self.partition_start.block
+                    && self.current_row.row < self.partition_start.row + (lag.offset as usize)
+                {
+                    let default_value = match lag.default.clone() {
+                        LagLeadDefault::Null => Scalar::Null,
+                        LagLeadDefault::Index(col) => {
+                            let block =
+                                &self.blocks[self.current_row.block - self.first_block].block;
+                            let col = block.get_by_offset(col).value.as_column().unwrap();
```
Comment (sundy-li) [inline src/query/service/src/pipelines/processors/transforms/window/transform_window.rs:501]:
```
let value = &block.get_by_offset(lag.arg).value;
value.index(self.current_row.row).unwrap().to_owned()
```

File: src/query/service/src/pipelines/processors/transforms/window/transform_window.rs [hunk 3]
```diff
@@ -490,28 +490,29 @@ impl<T: Number> TransformWindow<T> {
...
+                            let value = &block.get_by_offset(col).value;
+                            value.index(self.current_row.row).unwrap().to_owned()
+                        }
                     };
+                    default_value
+                } else {
+                    let block = self
+                        .blocks
+                        .get(self.frame_start.block - self.first_block)
+                        .cloned()
```
Comment (RinChanNOWWW) [inline src/query/service/src/pipelines/processors/transforms/window/transform_window.rs:510]: This clone is not need.

File: src/query/service/src/pipelines/processors/transforms/window/transform_window.rs [hunk 4]
```diff
@@ -490,28 +490,29 @@ impl<T: Number> TransformWindow<T> {
...
-                        let block = self
-                            .blocks
-                            .get(self.frame_start.block - self.first_block)
-                            .cloned()
-                            .unwrap()
-                            .block;
-                        let col = block.get_by_offset(lag.arg).value.as_column().unwrap();
-                        col.index(self.frame_start.row).unwrap().to_owned()
+                let lag_row = if self.current_row.block == self.partition_start.block
+                    && self.current_row.row < self.partition_start.row + (lag.offset as usize)
```
Comment (RinChanNOWWW) [inline src/query/service/src/pipelines/processors/transforms/window/transform_window.rs:494]: We can just check if `self.frame_start == self.frame_end` to judge if the frame is empty (or invalid).

**Review by sundy-li — APPROVED** [review]

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**